MIGRATION.portable-stories.md

```diff
@@ -0,0 +1,368 @@
+
+      stories.forEach(({ name, story }) => {
+        // Instead of not running the test, you can create logic to skip it, flagging it accordingly in the test results.
+        const testFn = story.parameters.storyshots?.skip ? test.skip : test;
+
+        testFn(name, async () => {
+          const mounted = render(story());
+          // Ensures a consistent snapshot by waiting for the component to render by adding a delay of 1 ms before taking the snapshot.
+          await new Promise((resolve) => setTimeout(resolve, 1));
+
+          expect(mounted.container).toMatchSnapshot();
+        });
+      });
+    });
+  });
+});
+```
+
+Running this example will generate a single snapshot file (i.e., `storybook.test.ts.snap`) with all the stories. However, if you were previously using a multi-snapshot configuration with the Storyshots addon, you can adjust the example above to include Vitest's [`toMatchFileSnapshot`](https://vitest.dev/guide/snapshot.html#file-snapshots) API. For example:
+
+```ts
+// ...Code omitted for brevity
+
+describe(options.suite, () => {
+  // 👇 Add storyDir in the arguments list
+  getAllStoryFiles().forEach(({ filePath, storyFile, storyDir }) => {
+    // ...Previously existing code
+    describe(title, () => {
+      // ...Previously existing code
+      stories.forEach(({ name, story }) => {
+        // ...Previously existing code
+        testFn(name, async () => {
+          // ...Previously existing code
+
+          // 👇 Define the path to save the snapshot to:
+          const snapshotPath = path.join(
+            storyDir,
+            options.snapshotsDirName,
+            `${componentName}${options.snapshotExtension}`
+          );
+          expect(mounted.container).toMatchFileSnapshot(snapshotPath);
+        });
+      });
+    });
+  });
+});
+```
+
+When the example above runs, it will generate individual snapshot files, one per story, using the following naming convention and location:
+
+```
+components/Button/Button.stories.ts
+components/Button/__snapshots__/Primary.storyshot
+components/Button/__snapshots__/Secondary.storyshot
+```
+
+#### Jest
+
+If you're using Jest as your testing framework, you can begin migrating your snapshot tests to Storybook's portable stories with the `composeStories` helper API by referring to the following example. You will need to modify the code in your `storybook.test.ts` file as follows:
+
+```ts
+// storybook.test.ts
+
+import path from 'path';
+import * as glob from 'glob';
+
+// Replace your-framework with one of the supported Storybook frameworks (react, vue3)
+import type { Meta, StoryFn } from '@storybook/your-framework';
+
+import { describe, test, expect } from '@jest/globals';
+
+// Replace your-testing-library with one of the supported testing libraries (e.g., react, vue)
+import { render } from '@testing-library/your-testing-library';
+
+// Adjust the import based on the supported framework or Storybook's testing libraries (e.g., react, testing-vue3)
```

Comment:
if people use `@storybook/react` they're good, but if they use `@storybook/testing-vue3`, they have to install the package. In the future it will be integrated into `@storybook/vue3`, but right now that's the case

Reply:
@yannbf I've pushed a commit that added a callout to this situation. As Vue 3 is the wildcard here, users must install and use it whether they are enabling `setProjectAnnotations` or using the `composeStories` API. I removed the mention of the integration to avoid confusion and have users ask us about the integration into the Vue 3 renderer as we don't have a precise date on it. I don't want to set expectations until that work is planned correctly and added.